src/utils/aggregation.jl

```diff
@@ -153,43 +217,116 @@ step = ["step-1","step-2", "step-3"] # has to match the date range
+    obj_value = Dict{Symbol, Any}(:OBJECTIVE_FUNCTION => optimizer[:obj_value])
+
+    if !isepmty(dual)
+        duals = _reading_references(duals, dual, stage, references, extra_time_length)
+        results = make_results(variables, obj_value, optimizer, time_stamp, duals)
+    else
+        results = make_results(variables, obj_value, optimizer, time_stamp)
+    end
+    if (:write in keys(kwargs)) == true
+        file_type = get(kwargs, :file_type, Feather)
+        write_model_results(results, normpath("$file_path/../../../../"),"results"; file_type = file_type)
+    end
+    return results
 end
 
-file_path = references[stage][variable[1]][1,:File_Path]
-opt_file_path = joinpath(dirname(file_path),"optimizer_log.feather")
-optimizer = Dict{Symbol, Any}(eachcol(Feather.read("$opt_file_path"),true))
-obj_value = Dict{Symbol, Any}(:OBJECTIVE_FUNCTION => optimizer[:obj_value])
-results = OperationsProblemResults(variable_dict, obj_value, optimizer, time_stamp)
-if (:write in keys(kwargs)) == true
-    file_type = get(kwargs, :file_type, Feather)
-write_results(results, normpath("$file_path/../../../../"),"results"; file_type = file_type)
+function _concat_string(duals::Vector{Symbol})
+    duals = (String.(duals)).*"_dual"
+    return duals
 end
-return results
+""" 
+    make_references(sim::Simulation, date_run::String)
+
+Creates a dictionary of variables with a dictionary of stages
+that contains dataframes of date/step/and desired file path
+so that the results can be parsed sequentially by variable
+and stage type.
+
+**Note:** make_references can only be run after run_sim_model
+or else, the folder structure will not yet be populated with results
+
+# Arguments
+-`sim::Simulation = sim`: simulation object created by Simulation()
+-`date_run::String = "2019-10-03T09-18-00"``: the name of the file created
+that contains the specific simulation run of the date run and "-test"
 
+# Example
+```julia
+sim = Simulation("test", 7, stages, "/Users/lhanig/Downloads/"; 
+verbose = true, system_to_file = false)
+run_sim_model!(sim::Simulation; verbose::Bool = false, kwargs...)
+references = make_references(sim, "2019-10-03T09-18-00-test")
+```
+"""
+function make_references(sim::Simulation, date_run::String; kwargs...)
+    sim.ref.date_ref[1] = sim.daterange[1]
+    sim.ref.date_ref[2] = sim.daterange[1]
+    references = Dict()
+    for (ix, stage) in enumerate(sim.stages)
+        variables = Dict()
+        interval = PSY.get_forecasts_interval(stage.sys)
+        variable_names = collect(keys(sim.stages[ix].canonical.variables))
+        if :dual_constraints in keys(kwargs) && !isnothing(get_constraints(stage.canonical))
+            dual_cons = _concat_string(kwargs[:dual_constraint])
+            variable_names = vcat(variable_names, dual_cons)
+        end
+        for name in variable_names
+            variables[variable_names[name]] = DataFrames.DataFrame(Date = Dates.DateTime[],
```

Comment:
variable_names is an Array and name is a symbol. This produces an error because you can't index an Array with a Symbol. What is the objective here?